lib/dry/types/rails/registry.rb

```diff
@@ -0,0 +1,15 @@
+module Dry
+  module Types
+    module Rails
+      class Registry < ::Dry::Container::Registry
+        def call(container, key, item, options)
+          key = key.to_s.dup.freeze
+          @_mutex.synchronize do
+            container.delete(key) if container.key?(key)
```

Comment:
@AMHOL Just removed the whole custom registry as it wasn't needed (forgot I had left it in) :)